src/wp-includes/blocks.php

```diff
@@ -1126,15 +1126,39 @@ function build_query_vars_from_query_block( $block, $page ) {
 			$query['offset']         = ( $per_page * ( $page - 1 ) ) + $offset;
 			$query['posts_per_page'] = $per_page;
 		}
-		if ( ! empty( $block->context['query']['categoryIds'] ) ) {
-			$term_ids              = array_map( 'intval', $block->context['query']['categoryIds'] );
-			$term_ids              = array_filter( $term_ids );
-			$query['category__in'] = $term_ids;
+		// Migrate `categoryIds` and `tagIds` to `tax_query` for backwards compatibility.
+		if ( ! empty( $block->context['query']['categoryIds'] ) || ! empty( $block->context['query']['tagIds'] ) ) {
```

Comment:
I can't see why this outer if is needed, the same checks are being done inside it.

Reply:
Just to  'encapsulate' (let's say) the 'migration' code. Also both old attributes merge into a single one and without the extra if we would have more checks about new value and merge, etc..

I think there is no harm to leave this as is.